crawl-ref/source/skills.cc

```diff
@@ -1145,7 +1146,7 @@ pair<int,int> skill_level_to_diffs(skill_type skill, double amount, int scaled_t
         you_skill += ash_skill_point_boost(skill, you.skills[skill] * 10);
     }
     if (target == you_skill)
-        return make_pair(0, 0);
+        return skill_diff();
 
     // XXX: reducing with ash bonuses in play could lead to weird results.
     const bool reduced = target < you_skill;
```

Comment:
`reduced` seems to have significant impact on a number of decisions throughout this function, but it's not clear from its name what is being reduced. From looking at how it's defined I still don't really know why `target < you_skill` is meaningful here, or why the changes it makes later make sense. Are we trying to compute...how much experience I would have to lose in order to no longer be at mindelay, or something weird like that? A more descriptive name and/or a lengthier comment explaining what it means would be welcome.